src/Components/WorkloadsListTable/WorkloadsListTable.js

```diff
@@ -178,15 +187,28 @@ const WorkloadsListTable = ({
     },
   };
 
+  const onSetPage = (_e, pageNumber) => {
+    setRowsFiltered(false);
+    const newOffset = pageNumber * filters.limit - filters.limit;
+    updateFilters({ ...filters, offset: newOffset });
+  };
+
+  const onSetPerPage = (_e, perPage) => {
+    //THIS IS A DUMMY QUERY THAT WILL BE REMOVED WHEN WE WORK ON CONNECTING FILTERS/SORTING/PAGES TO THE URL PARAMS
+    setTempQuery(Math.random());
```

Comment:
You don't need this. In fact, the thing is that if you select the same per page value and it doesn't reload the page correctly is the indication of the bug in the current implementation and it has a chance to persist in the future.

The reason why it happens is that useEffect is not triggered because perPage has the same value (it is primitive and useEffect doesn't compare references like with objects, but just compares the value). Thus, setRowsFiltered(true) is not called neither. One of the safe ways to fix this is to not to call updateFilters in these callbacks if perPage or pageNumber are the same as currently applied values. The table will not go into the loading state and still show the same results which is correct.

Reply:
Fixed this bug